From 395982afa822769f7557da434c560cb6f231e69d Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:17:41 +0200
Subject: [PATCH 22/46] AnnaThermostat: add support for central_heating_state

Parse central_heating_state and publish it as an on/off switch sensor.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 225636577..b86cf637e 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1175,6 +1175,14 @@ void CAnnaThermostat::GetMeterDetails()
 					SendSwitch(applianceIDhex, 7, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
 				}
 			}
+			else if (sname == "central_heating_state")
+			{
+				if (!tmpstr.empty())
+				{
+					bool isOn = (strcmp(tmpstr.c_str(), "on") == 0);
+					SendSwitch(applianceIDhex, 8, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
+				}
+			}
             else if (sname == "central_heater_water_pressure")
             {
                 if (!tmpstr.empty())
